File: tests/fixtures/safe_secret_names.js
// Benign names containing secret-keyword substrings, or low-signal
// keywords with env-sourced values. None should be flagged by
// js/no-hardcoded-secret after the word-boundary + value-gate fix.

// Substring false positives.
const author = "Pallets";
const authors = "core team";
const authenticated = "yes";
const authorizationScheme = "Bearer";
const tokenizer = "bert-base-uncased";
const secretarialNote = "filed";

// Low-signal keyword values that are clearly not secrets.
const auth = "/login";
const authUrl = "https://example.com/oauth";
const token = "see docs for the token format";

// Env-sourced secret-named values (not hardcoded literals).
const password = process.env.PW;
const apiKey = process.env.API_KEY;
const secretKey = process.env.SECRET_KEY;

module.exports = {
  author,
  authors,
  authenticated,
  authorizationScheme,
  tokenizer,
  secretarialNote,
  auth,
  authUrl,
  token,
  password,
  apiKey,
  secretKey,
};

File: tests/fixtures/safe_secret_names.py
# Benign names that contain secret-keyword *substrings* or low-signal
# secret keywords with non-secret values. None of these should be flagged
# by `py/no-hardcoded-secret` after the word-boundary + value-gate fix.
import os

# Substring false positives: these identifiers merely *contain* a keyword.
author = "Pallets"
authors = "Armin Ronacher"
authored_by = "core team"
authenticated = True
authentication_scheme = "Bearer"
authorization_header = "X-Custom"
tokenizer = "bert-base-uncased"
tokenize_input = "whitespace"
secretarial_note = "filed"

# Low-signal keywords (auth / token) whose values are clearly not secrets
# (URL / path / value-with-spaces -> rejected by looks_like_secret_value).
auth = "/login"
auth_url = "https://example.com/oauth"
token = "see docs for the token format"

# Env-sourced "secret" names — the value is not a hardcoded literal.
password = os.environ.get("PW", "")
api_key = os.getenv("API_KEY")
secret_key = os.environ["SECRET_KEY"]
credentials = os.environ.get("CREDS")
